traffic_router/ultimate-test-harness/dns_test.go

```diff
@@ -0,0 +1,136 @@
+		if trafficRouter.DSHost[len(trafficRouter.DSHost)-1] != '.' {
+			trafficRouter.DSHost += "."
+		}
+		for ipAddressIndex, ipAddress := range trafficRouter.IPAddresses {
+			isIPv4 := strings.Contains(ipAddress, ".")
+			if trafficRouter.ClientIP == "" && trafficRouter.ClientIPAddressMap.Zones != nil {
+				if benchmark.CoverageZoneLocation != "" {
+					location := trafficRouter.ClientIPAddressMap.Map[benchmark.CoverageZoneLocation]
+					trafficRouter.ClientIP = location.GetFirstIPAddressOfType(isIPv4)
+				}
+				if trafficRouter.ClientIP == "" {
+					for _, location := range trafficRouter.ClientIPAddressMap.Map {
+						trafficRouter.ClientIP = location.GetFirstIPAddressOfType(isIPv4)
+						if trafficRouter.ClientIP != "" {
+							break
+						}
+					}
+				}
+			}
+
+			answers, failures := 0, 0
+			redirectsChannels := make([]chan int, benchmark.ThreadCount)
+			failuresChannels := make([]chan int, benchmark.ThreadCount)
+			for threadIndex := 0; threadIndex < benchmark.ThreadCount; threadIndex++ {
+				redirectsChannels[threadIndex] = make(chan int)
+				failuresChannels[threadIndex] = make(chan int)
+				go benchmark.Run(t, redirectsChannels[threadIndex], failuresChannels[threadIndex], trafficRouterIndex, ipAddressIndex)
+			}
+
+			for threadIndex := 0; threadIndex < benchmark.ThreadCount; threadIndex++ {
+				answers += <-redirectsChannels[threadIndex]
+				failures += <-failuresChannels[threadIndex]
+			}
+			protocol := "IPv6"
+			if isIPv4 {
+				protocol = "IPv4"
+			}
+			var passed string
+			requestsPerSecond := answers / benchmark.BenchmarkSeconds
+			if requestsPerSecond > benchmark.RequestsPerSecondThreshold {
+				passedTests++
+				passed = "Yes"
+			} else {
+				failedTests++
+				passed = "No"
+			}
+			fmt.Fprintf(writer, "%s\t%s\t%s\t%s\t%d\t%d\t%d\n", trafficRouter.Hostname, protocol, trafficRouter.DSHost, passed, requestsPerSecond, answers, failures)
+			writer.Flush()
+		}
+	}
+	summary := fmt.Sprintf("%d out of %d load tests passed", passedTests, passedTests+failedTests)
+	if failedTests < 1 {
+		t.Logf(summary)
+	} else {
+		t.Fatal(summary)
+	}
+}
+
+func (b DNSBenchmark) Run(t *testing.T, answersChannel chan int, failuresChannel chan int, trafficRouterIndex int, ipAddressIndex int) {
+	stopTime := time.Now().Add(time.Duration(b.BenchmarkSeconds) * time.Second)
+	answers, failures := 0, 0
+	client := dns.Client{Net: "udp", Timeout: 10 * time.Second}
+	message := new(dns.Msg)
+	trafficRouter := b.TrafficRouters[trafficRouterIndex]
+	address := trafficRouter.IPAddresses[ipAddressIndex] + ":53"
+
+	message.SetQuestion(trafficRouter.DSHost, dns.TypeA)
+	for time.Now().Before(stopTime) {
+		r, _, err := client.Exchange(message, address)
+		if err == nil && len(r.Answer) > 0 {
+			answers++
+		} else {
+			failures++
+		}
+	}
+	answersChannel <- answers
+	failuresChannel <- failures
+}
+
+func TestDNSLoad(t *testing.T) {
+	var trafficRouterDetails []TRDetails = GetTrafficRouterDetails(t)
```

Comment:
You can get rid of the `[]TRDetails` here.

Reply:
Got rid of `[]TRDetails` in 6cee5e1042